### PR TITLE
add optional secureJsonData to grafana datasources

### LINKED ISSUE #295: Add custom headers to Grafana Datasource
### Enhancement Proposal

Currently, the Grafana Datasource library sets up a new datasource through a URL and port; however, some services (e.g., Mimir) need to pass custom headers as well (e.g., a multitenancy org id: even if not using one, Mimir requires the header to be set to a specific string).

We should add the ability to pass a dictionary of headers to `GrafanaSourceProvider`.

### PR DESCRIPTION
**Pull request by lucabello**
## Issue
Closes #295.

Some datasources, like Mimir, need custom HTTP Headers to be specified in the datasource configuration. 

As seen [here](https://grafana.com/docs/grafana/latest/administration/provisioning/#custom-http-headers-for-data-sources), header names can be specified under `jsonData`, which we currently build from the `extra_data` parameter in the `GrafanaSourceProvider` constructor.

However, the value for a header must be specified under `secureJsonData`, which we currently don't allow to specify in the `grafana_source` library.

## Solution
This PR adds an optional `secure_extra_data` constructor argument, right next to `extra_data`, so that we can (in the same fashion as `jsonData`) specify `secureJsonData` for a datasource.

### CODE REVIEW

**Comment by lucabello** [comment]
Headers are set successfully now:
Screenshot_20240206_144103

And the Mimir datasource is happy as a consequence:
Screenshot_20240206_144037

File: lib/charms/grafana_k8s/v0/grafana_source.py [hunk 1]
```diff
@@ -32,6 +32,8 @@
     `refresh_event`: A `PebbleReady` event from `charm`, used to refresh
         the IP address sent to Grafana on a charm lifecycle event or
         pod restart
+    `extra_fields`: None
```
Comment (PietroPasotti) [inline lib/charms/grafana_k8s/v0/grafana_source.py:35]: I'd like to see in the doc what these are for / what they map to, in terms of datasource config

**Review by PietroPasotti — CHANGES_REQUESTED** [review]

File: src/charm.py [hunk 2]
```diff
@@ -1131,6 +1131,8 @@ def _generate_datasource_config(self) -> str:
             }
             if source_info.get("extra_fields", None):
                 source["jsonData"] = source_info.get("extra_fields")
+            if source_info.get("secure_extra_fields", None):
+                source["secureJsonData"] = source_info.get("secure_extra_fields")
```
[the remark this answers is not in the document]
Comment (PietroPasotti) [reply src/charm.py:1135]: (not something you NEED to fix in this PR unless you agree we should rename both vars, but maybe let's put a wishlist/todo for it?)

[the remark this answers is not in the document]
Comment (lucabello) [reply src/charm.py:1135]: I have no idea why that was the case :) I'll open a new issue to refactor it to be more clear!

File: lib/charms/grafana_k8s/v0/grafana_source.py (hunk 1 above)
Comment (lucabello) [reply lib/charms/grafana_k8s/v0/grafana_source.py:35]: It's already in the class documentation; this part of the docs only specifies the default argument (note how that's true for all others except `refresh_event`)

**Review by PietroPasotti — APPROVED** [review]
looks grafana to me

**Pull request merged**